### PR TITLE
Adds recently added storage and signers related artifacts to Chains Spec

### PR DESCRIPTION
**Pull request by PuneetPunamiya**
Following are the artifacts added

  - storage.grafeas.notehint
  - signers.x509.identity.token.file
  - signers.x509.identity.mirror.url

Signed-off-by: Puneet Punamiya ppunamiy@redhat.com

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Run `make test lint` before submitting a PR
- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

## Breaking Change
* Chains have added one more field value called as `manual` 
for the field `transparency.enabled`, hence changing the type 
from `bool` to `string`

### CODE REVIEW

File: pkg/apis/operator/v1alpha1/tektonchain_types.go [hunk 1]
```diff
@@ -97,7 +100,7 @@ type Chain struct {
 	KMSAuthSpireAudience string `json:"signers.kms.auth.spire.audience,omitempty"`
 
 	// transparency config
-	TransparencyConfigEnabled *bool  `json:"transparency.enabled,omitempty"`
+	TransparencyConfigEnabled string `json:"transparency.enabled,omitempty"`
```
Comment (piyush-garg) [inline pkg/apis/operator/v1alpha1/tektonchain_types.go:103]: as we are changing the type here, do we need to handle any conversions?

File: pkg/apis/operator/v1alpha1/tektonchain_validation.go [hunk 2]
```diff
@@ -90,5 +94,25 @@ func (tcs *TektonChainSpec) ValidateChainConfig(path string) (errs *apis.FieldEr
...
+		for i, v := range input {
+			input[i] = strings.TrimSpace(v)
+			if !allowedX509SignerFulcioProvider.Has(input[i]) {
+				errs = errs.Also(apis.ErrInvalidValue(input[i], path+".signers.x509.fulcio.provider"))
+			}
+		}
+	}
+
+	if tcs.TransparencyConfigEnabled != "" {
+		input := strings.Split(tcs.TransparencyConfigEnabled, ",")
```
Comment (piyush-garg) [inline pkg/apis/operator/v1alpha1/tektonchain_validation.go:108]: why we need to do this? this is not the single value field ?

File: pkg/apis/operator/v1alpha1/tektonchain_validation.go [hunk 3]
```diff
@@ -90,5 +94,25 @@ func (tcs *TektonChainSpec) ValidateChainConfig(path string) (errs *apis.FieldEr
 		}
 	}
 
+	if tcs.X509SignerFulcioProvider != "" {
```
Comment (piyush-garg) [inline pkg/apis/operator/v1alpha1/tektonchain_validation.go:97]: why we need to do this? this is not the single value field ?

**Review by jkandasa — APPROVED** [review]
LGTM

File: pkg/apis/operator/v1alpha1/tektonchain_validation.go [hunk 4]
```diff
@@ -25,7 +25,12 @@ import (
 	"knative.dev/pkg/apis"
 )
 
-var allowedArtifactsStorage = sets.NewString("tekton", "oci", "gcs", "docdb", "grafeas", "kafka")
+var (
+	allowedArtifactsTaskRunFormat    = sets.NewString("in-toto", "slsa/v1")
+	allowedArtifactsStorage          = sets.NewString("tekton", "oci", "gcs", "docdb", "grafeas", "kafka")
+	allowedX509SignerFulcioProvider  = sets.NewString("google", "spiffe", "github", "filesystem")
+	allowedTransparencyConfigEnabled = sets.NewString("true", "false", "manual")
```
Comment (jkandasa) [inline pkg/apis/operator/v1alpha1/tektonchain_validation.go:32]: Seems all these sets allowed empty string, can we add empty string in the allowed sets and omit `if xyz != ""` statement on the validation?

**Comment by vdemeester** [comment]
@PuneetPunamiya Does it "complete" https://github.com/tektoncd/operator/pull/1389 ?

**Comment by PuneetPunamiya** [comment]
> @PuneetPunamiya Does it "complete" #1389 ?

you mean does this pr adds pipelinerun artifacts which are added in #1389

**Comment by vdemeester** [comment]
> > @PuneetPunamiya Does it "complete" #1389 ?
> 
> you mean does this pr adds pipelinerun artifacts which are added in #1389

More the opposite 🙃. We should make sure we get #1389 in, in addition to this PR.

**Comment by PuneetPunamiya** [comment]
> > > @PuneetPunamiya Does it "complete" #1389 ?
> > 
> > 
> > you mean does this pr adds pipelinerun artifacts which are added in #1389
> 
> More the opposite upside_down_face. We should make sure we get #1389 in, in addition to this PR.

yep sure, have not addressed artifacts which are already added in that pr👍🏻

File: pkg/apis/operator/v1alpha1/tektonchain_validation.go [hunk 5]
```diff
@@ -25,7 +25,13 @@ import (
...
 )
 
-var allowedArtifactsStorage = sets.NewString("tekton", "oci", "gcs", "docdb", "grafeas", "kafka")
+var (
+	allowedArtifactsTaskRunFormat     = sets.NewString("in-toto", "slsa/v1", "")
+	allowedArtifactsPipelineRunFormat = sets.NewString("in-toto", "slsa/v1", "")
+	allowedArtifactsStorage           = sets.NewString("tekton", "oci", "gcs", "docdb", "grafeas", "kafka")
+	allowedX509SignerFulcioProvider   = sets.NewString("google", "spiffe", "github", "filesystem", "")
+	allowedTransparencyConfigEnabled  = sets.NewString("true", "false", "manual", "")
+)
```
Comment (jkandasa) [inline pkg/apis/operator/v1alpha1/tektonchain_validation.go:34]: in all sets add the empty string `""` as a first value. will be more readable
and move `allowedArtifactsStorage` to last on the var list, as it do not contain `""` value

Comment (PuneetPunamiya) [reply pkg/apis/operator/v1alpha1/tektonchain_validation.go:34]: done !

**Review by jkandasa — APPROVED** [review]

**Review by vdemeester — APPROVED** [review]

**Pull request merged**